src/Microsoft.DotNet.Interactive.Formatting/Formatter.cs

```diff
@@ -766,4 +685,40 @@ private static bool TryGetPropertyValue<T>(object fromObject, string propertyNam
         value = default;
         return false;
     }
+
+    internal static bool ShouldIncludePropertiesInOutput(this Type type)
+    {
+        if (type.IsArray)
+        {
+            return false;
+        }
+
+        if (type.IsNestedPrivate) //e.g. RangeIterator
+        {
+            return false;
+        }
+
+        if (typeof(ICollection).IsAssignableFrom(type))
```

Comment:
This may be a micro-optimization, but considering that we are already looking at implemented interfaces in the loop below, would it be better to add an early out for `ICollection` there instead? (Taking a quick look at the implementation for `IsAssignableFrom` it seems to be doing a bunch morethings before checking if an implemented interface `IsSubClassOf(typeof(ICollection))` )